### PR TITLE
Wagtail 5.1 - upgrade considerations

### PR DESCRIPTION
**Pull request by katdom13**
[Wagtail 5.1 release notes](https://docs.wagtail.org/en/stable/releases/5.1.html)

Changes:
- [wagtail.contrib.modeladmin is deprecated
](https://docs.wagtail.org/en/stable/releases/5.1.html#wagtail-contrib-modeladmin-is-deprecated)

Upstream PR: https://github.com/wagtail/wagtail-personalisation/pull/250

### CODE REVIEW

File: sandbox/sandbox/settings.py [hunk 1]
```diff
@@ -53,7 +54,7 @@
     "wagtail.search",
     "wagtail.admin",
     "wagtail",
-    "wagtail.contrib.modeladmin",
+    "wagtail_modeladmin" if WAGTAIL_VERSION >= (5, 1) else "wagtail.contrib.modeladmin",
     "wagtailfontawesome",
```
Comment (katdom13) [inline sandbox/sandbox/settings.py]: I wonder if we have to update this to `wagtailfontawesomesvg` instead

File: setup.py
Comment (nickmoreton) [inline setup.py:1]: Can you add `"Framework :: Wagtail :: 5"` in the classifiers please

File: sandbox/sandbox/settings.py (hunk 1 above)
Comment (nickmoreton) [reply sandbox/sandbox/settings.py]: I don't know for sure. It makes sense though. Can you test it out using the sandbox app?

**Review by nickmoreton — CHANGES_REQUESTED** [review]
I also wonder if the comment I posted for wagtail-robots is also relevant here?

File: src/wagtail_personalisation/templates/modeladmin/wagtail_personalisation/segment/dashboard.html [hunk 2]
```diff
@@ -29,6 +29,7 @@ <h2>{{ segment }}</h2>
                                             <ul class="inspect segment_stats">
                                                 <li class="stat_card">
                                                     {% trans "This segment has been visited" %}
+                                                    <!-- TODO: Change this icon -->
                                                     <span class="icon icon-fa-rocket">{{ segment.visit_count|localize }} {% trans "time" %}{{ segment.visit_count|pluralize }}</span>
```
Comment (katdom13) [inline src/wagtail_personalisation/templates/modeladmin/wagtail_personalisation/segment/dashboard.html]: Hi @nickmoreton 
Any idea on how to use the icons? I already tried the following:
```suggestion
<!-- TODO: Change this icon -->
{% icon name="rocket" %}{{ segment.visit_count|localize }} {% trans "time" %}{{ segment.visit_count|pluralize }}
```

But it looks like:
image

I tried applying the class:
`{% icon name="rocket" classname="icon icon-rocket" %}`

But there are no styles being applied.

File: setup.py
Comment (katdom13) [reply setup.py:1]: Added here: https://github.com/torchbox-forks/wagtail-personalisation/pull/6/files#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R74

File: src/wagtail_personalisation/views.py
Comment (nickmoreton) [inline src/wagtail_personalisation/views.py:1]: I noticed the side bar icon was missing.

```
@modeladmin_register
class SegmentModelAdmin(ModelAdmin):
    """The model admin for the Segments administration interface."""

    model = Segment
    index_view_class = SegmentModelIndexView
    dashboard_view_class = SegmentModelDashboardView
    delete_view_class = SegmentModelDeleteView
    menu_icon = "rocket" <-- add this
```

File: src/wagtail_personalisation/templates/modeladmin/wagtail_personalisation/segment/dashboard.html (hunk 2 above)
Comment (nickmoreton) [reply src/wagtail_personalisation/templates/modeladmin/wagtail_personalisation/segment/dashboard.html]: I found this works:

```
<span>{% icon name="rocket" classname="w-w-4 w-h-4" %} {{ segment.visit_count|localize }} {% trans "time" %}{{ segment.visit_count|pluralize }}</span>
```

I found the classname options by inspecting the info button in the top bar of the HomePage listing and tried them.

```
<svg class="icon icon-info-circle w-w-4 w-h-4 group-hover:w-transform group-hover:w-scale-110" aria-hidden="true"><use href="#icon-info-circle"></use></svg>
```

Comment (nickmoreton) [reply src/wagtail_personalisation/templates/modeladmin/wagtail_personalisation/segment/dashboard.html]: I also tried this for the calendar icon:

```
<span>{% icon name="calendar-alt" classname="w-w-4 w-h-4" %} {{ segment.enable_date|days_since:segment.disable_date }} {% trans "day" %}{{ segment.enable_date|days_since:segment.disable_date|pluralize }}</span>
```
But thats a built in wagtail icon so didn't need to add to the register_icons method. I wonder if there are enough suitable icons to be able to not even use wagtailfontawesomesvg

File: src/wagtail_personalisation/views.py
Comment (nickmoreton) [reply src/wagtail_personalisation/views.py:1]: Actually rocket may not be the correct icon, it's a snowflake on the screen shot.

**Review by nickmoreton — COMMENTED** [review]
Hi @katdom13 I think I've found the solution and added comments, what do you think.

**Comment by katdom13** [comment]
Hi @nickmoreton ,
Thank you for your suggestions! They are working.

I still had to register those icons that are not in the styleguide, but managed to find suitable alternatives for the rest.

New commits start here: https://github.com/torchbox-forks/wagtail-personalisation/pull/6/commits/a173aa4b8f0459c77c39530be903505fc09b8ec9

image

**Review by nickmoreton — APPROVED** [review]
Hi @katdom13 Excellent, I approve.

**Pull request merged**